core/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala

```diff
@@ -217,6 +217,41 @@ class SQLPlanParserSuite extends BaseTestSuite {
     }
   }
 
+  test("Parse Execs within WholeStageCodeGen in Order") {
```

Comment:
The unit-test fails on different spark-versions
I tested it on spark-3.40

build cmd:
```
mvn -Dbuildver=340 package
```

failed tests

```
- Parse Execs within WholeStageCodeGen in Order *** FAILED ***
  ArrayBuffer(exec: AdaptiveSparkPlan, expr: , sqlID: 201 , speedupFactor: 1.0, duration: None, nodeId: 0, isSupported: false, children: , stages: , shouldRemove: false, exec: WholeStageCodegen (2), expr: WholeStageCodegen (2), sqlID: 201 , speedupFactor: 4.41, duration: Some(3), nodeId: 1, isSupported: true, children:
         exec: HashAggregate, expr: , sqlID: 201 , speedupFactor: 4.41, duration: None, nodeId: 2, isSupported: true, children: , stages: 2, shouldRemove: false, stages: 2, shouldRemove: false, exec: AQEShuffleRead, expr: , sqlID: 201 , speedupFactor: 3.0, duration: None, nodeId: 3, isSupported: true, children: , stages: , shouldRemove: false, exec: Exchange, expr: , sqlID: 201 , speedupFactor: 4.9, duration: Some(279), nodeId: 4, isSupported: true, children: , stages: 0,2, shouldRemove: false, exec: WholeStageCodegen (1), expr: WholeStageCodegen (1), sqlID: 201 , speedupFactor: 2.8, duration: Some(321), nodeId: 5, isSupported: true, children:
         exec: HashAggregate, expr: , sqlID: 201 , speedupFactor: 4.41, duration: None, nodeId: 6, isSupported: true, children: , stages: 0, shouldRemove: false
         exec: Expand, expr: , sqlID: 201 , speedupFactor: 3.0, duration: None, nodeId: 7, isSupported: true, children: , stages: 0, shouldRemove: false
         exec: LocalTableScan, expr: , sqlID: 201 , speedupFactor: 1.0, duration: None, nodeId: 8, isSupported: false, children: , stages: 0, shouldRemove: false, stages: 0, shouldRemove: false) had size 5 instead of expected size 3 (SqlPlanParserSuite.scala:237)
```